File: 12Jan2023.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int f1 = 0, f2 = 0;
        for (int i = 0; i < s.length(); i++) {
            if (isVowel(s.charAt(i))) {
                if (i < s.length() / 2) {
                    f1++;
                } else {
                    f2++;
                }
            }
        }
        return f1 == f2;

    }

    public boolean isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'A' || c == 'E' || c == 'I' || c == 'O'
                || c == 'U';
    }
}

File: 12Jan2023.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        f1=0;
        f2=0;
        for i in range(0,len(s)) :
            if(self.isVowel(s[i])) :
                if i<len(s)/2 :
                    f1+=1;
                else :
                    f2+=1;
        return f1==f2;
        
    
    def isVowel(self, c: chr) -> bool:
        return c=='a' or c=='e' or c=='i' or c=='o' or c=='u' or c=='A' or c=='E' or c=='I' or c=='O' or c=='U';